### PR TITLE
feat: restricted api support

### PR DESCRIPTION
**Pull request by AuHau**

### CODE REVIEW

File: scripts/bee.sh [hunk 1]
```diff
@@ -149,6 +157,12 @@ do
         BEE_PASSWORD="${1#*=}"
         shift 1
         ;;
+        --restrict=*)
+        RESTRICTED="true"
+        RESTRICTED_PASSWORD="${1#*=}"
+        RESTRICTED_PASSWORD_HASH=$(htpasswd -bnBC 10 "" $RESTRICTED_PASSWORD | tr -d ':\n')
```
Comment (nugaon) [inline scripts/bee.sh:163]: it requires to install `apache2-utils`.

can we generate the password hash with `openssl passwd` or some other tool that is available by default?

File: scripts/bee.sh [hunk 2]
```diff
@@ -278,9 +297,15 @@ echo "Check whether the queen node has been connected to every worker..."
 ELAPSED_TIME=0
 WAITING_TIME=2
 TIMEOUT=$((2*30*WAITING_TIME))
+RESTRICTED_TOKEN=""
 while (( TIMEOUT > ELAPSED_TIME )) ; do
     check_queen_is_running
-    COUNT=$(count_connected_peers)
+    if [ $RESTRICTED=="true" -a -z $RESTRICTED_TOKEN ]; then
```
Comment (nugaon) [inline scripts/bee.sh:303]:
```suggestion
    if [ $RESTRICTED == "true" ] && [ -z "$RESTRICTED_TOKEN" ]; then
```

File: scripts/environment.sh [hunk 3]
```diff
@@ -86,6 +89,13 @@ do
         HOSTNAME="${1#*=}"
         shift 1
         ;;
+        --restrict*)
+        RESTRICTED=true
+        if [ "${1#*=}" != "--restrict" ] ; then
+          RESTRICTED_PASSWORD="${1#*=}"
+        fi
```
Comment (nugaon) [inline scripts/environment.sh:96]: I don't completely get this part...

the `restrict` should come with required parameter I think, and you should check `--restrict=*` as at the other options.

I don't see any yield of defining default value for the restricted mode when it is optional anyway. Moreover, there is no documentation about this default password which would be necessary.

File: scripts/bee.sh [hunk 4]
```diff
@@ -198,6 +212,9 @@ if [ -z "$QUEEN_CONTAINER_IN_DOCKER" ] || $EPHEMERAL ; then
     fi
 
     echo "start Bee Queen process"
+    if [ $RESTRICTED=="true" ]; then
```
Comment (nugaon) [inline scripts/bee.sh:215]:
```suggestion
    if [ $RESTRICTED == "true" ]; then
```

File: scripts/bee.sh [hunk 5]
```diff
@@ -278,9 +297,15 @@ echo "Check whether the queen node has been connected to every worker..."
 ELAPSED_TIME=0
 WAITING_TIME=2
 TIMEOUT=$((2*30*WAITING_TIME))
+RESTRICTED_TOKEN=""
 while (( TIMEOUT > ELAPSED_TIME )) ; do
     check_queen_is_running
-    COUNT=$(count_connected_peers)
+    if [ $RESTRICTED=="true" -a -z $RESTRICTED_TOKEN ]; then
+      RESTRICTED_TOKEN=$(get_token $RESTRICTED_PASSWORD)
```
Comment (nugaon) [inline scripts/bee.sh:304]:
```suggestion
      RESTRICTED_TOKEN=$(get_token "$RESTRICTED_PASSWORD")
```

File: scripts/bee.sh [hunk 6]
```diff
@@ -278,9 +297,15 @@ echo "Check whether the queen node has been connected to every worker..."
...
+      echo "Fetched Bearer token: $RESTRICTED_TOKEN"
+    fi;
+
+    COUNT=$(count_connected_peers $RESTRICTED_TOKEN)
```
Comment (nugaon) [inline scripts/bee.sh:308]:
```suggestion
    COUNT=$(count_connected_peers "$RESTRICTED_TOKEN")
```

**Review by nugaon — CHANGES_REQUESTED** [review]

File: scripts/bee.sh (hunk 1 above)
Comment (AuHau) [reply scripts/bee.sh:163]: Yeah, you are right, but I have not discovered any other way. I don't think `openssl` supports `bcrypt`. I have asked if Bee team could implement simple CLI command that would provide this hashing functionality, but until then I would go with this...

File: scripts/bee.sh (hunk 2 above)
Comment (AuHau) [reply scripts/bee.sh:303]: Why you don't want to use `-a`?

File: scripts/environment.sh (hunk 3 above)
Comment (AuHau) [reply scripts/environment.sh:96]: > I don't see any yield of defining default value for the restricted mode when it is optional anyway. Moreover, there is no documentation about this default password which would be necessary.

Ehm, its here: https://github.com/ethersphere/bee-factory/pull/69/files#diff-f3736fd0d9f6c379e7622c6230d742bb019c1120223e1c3857bb1808d27d5413R11

Comment (nugaon) [reply scripts/environment.sh:96]: I meant in the README...

File: scripts/bee.sh (hunk 2 above)
Comment (nugaon) [reply scripts/bee.sh:303]: because it is ugly and also the ShellCheck underlines.

File: scripts/bee.sh (hunk 1 above)
Comment (nugaon) [reply scripts/bee.sh:163]: Okay, then please add this as a dependency to the project until then

Comment (AuHau) [reply scripts/bee.sh:163]: I assume by "add this as a dependency" you mean to document that in README right?

**Pull request merged**